### ISSUE TITLE
Install Issue on Armbian [on NanoPi M4]

### ISSUE THREAD

**Issue by github59173** [iiab/iiab#3655]
### Expected behavior and actual behavior: 
Expected behavior: Complete install of IIAB 8.1.
Actual behavior: Script forcibly closed and nanopi-m4 resets after update check is complete and asks for any key press during install.

### Steps to reproduce the problem: 
Run the installer and get to the step after the installer has checked for update and has confirmed their are none to be installed before proceeding with the rest of the install after hitting any key to continue. After any key is pressed the script is closed and the nanopi-m4 is forcibly restarted. (This step prompt to use the "sudo iiab" command if the script stops, which it does, then it force restarts the nanopi-m4 again).

### IIAB version, OS version, hardware details: 
IIAB 8.1, Armbian 23.8 Bookworm Cinnamon (and XFCE) desktop versions (https://www.armbian.com/nanopi-m4/), Nanopi-M4

### Please run the `iiab-diagnostics` command, to generate a URL summarizing your IIAB configuration, for volunteers seeking to help you: (Include that short URL here, Thanks!) 
"iiab-diagnostics: command not found" The software suite was not fully installed so the command doesn't exist.

**Comment by holta** [comment]
@github59173 thanks for explaining.

IIAB does not support Armbian at the moment.

Can you consider any of the following OS's?
https://wiki.iiab.io/go/FAQ#What_OS_should_I_use%3F

**Comment by github59173** [comment]
Unfortunately Nanipi-M4 official images are not up-to-date with Debian 12, or Ubuntu 23.10/22.04, and can't run Raspberry Pi OS as far as I know. If you can make a recommendation of a OS image that will work with a Nanopi-M4, that would be a HUGE help!

**Comment by holta** [comment]
Any idea when they might support Debian 12 "Bookworm" from 2023-06-10 which is probably your best bet?!

(Apologies I don't personally know much about the "NanoPi M4" !)

**Comment by github59173** [comment]
Last image update was September 14th. 2023 on google drive. I have no information if they will update the images to Debian 12 at any point. Unless I'm missing an compatible image in the list of available official images, I went the route of getting third party images with the close approximation to the requirements (Armbian Bookworm, pulled from official armbian website). Here are a couple of links to their tools and images if you want to take a look.
Images and Utli: https://dl.friendlyelec.com/nanopim4 
Wiki: https://wiki.friendlyelec.com/wiki/index.php/Main_Page#Embedded_Operating_Systems

**Comment by holta** [comment]
@github59173

In the interim until Debian 12 "Bookworm" is released for NanoPi M4, have you tried Debian 11 "Bullseye" ?

(It just might possibly work — or at least show possibilities.)

**Comment by github59173** [comment]
I was unsuccessful with using their official Debian 11 images. Many errors including locked package versions, also their images don't use Debian main repository, it uses their mirror tencent repositories. Is there a way to see what is causing the script to fail on Armbian builds? If some code requires tweaking for checks I don't mind doing that. I'd much prefer using an image that doesn't have all friendlyelec extra crap on it anyways.

**Comment by holta** [comment]
> Is there a way to see what is causing the script to fail on Armbian builds?

Can you run the steps below on a fresh/clean OS?
If something fails consistently/insurmountably, reply here with scrollback/screenshots?

```
apt update
apt -y dist-upgrade
apt -y install git
reboot

mkdir -p /etc/iiab /opt/iiab
cd /opt/iiab
git clone https://github.com/iiab/iiab

cd /opt/iiab/iiab
cp vars/local_vars_small.yml /etc/iiab/local_vars.yml
scripts/ansible
./install-iiab
```

Run all above as user `root` — or equivalently using sudo.

> If some code requires tweaking for checks I don't mind doing that.

Great if you can can:

Specifically, you'll later need to add a line specifying your OS and its major version number within [Lines 82-96 of /opt/iiab/iiab/scripts/local_facts.fact](https://github.com/iiab/iiab/blob/master/scripts/local_facts.fact#L82-L96) — definitely let us know how you make out there and/or if you need assistance when you reach that stage :pray:

**Comment by jvonau** [comment]
related #3298

**Comment by holta** [comment]
> related #3298

Indeed.  Useful tips for anybody interested in working on OS's like DietPi and/or Armbian:

- #3298
